ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,85 @@ def test_add_host_with_null_password(self, host):
         host.track_create()
         host.check_create(result)
 
+    @staticmethod
+    def modify_config_maxhostname(host_tracker, value):
+        command = host_tracker.make_command(
+            'config_mod',
+            **dict(
+                setattr=u'ipamaxhostnamelength={}'.format(value)))
+        command()
+
+    @staticmethod
+    def generate_hostname(total_length):
+        if total_length < 9:
+            raise ArithmeticError("Total DNS length in theses tests"
+                                  "must be at least 9")
+        return "aaa.bbb.{}".format((total_length - 8) * "c")
+
+    def test_config_maxhostname_invalid(self):
+        """Change config maxhostname to an invalid value
+        (lower than MAXHOSTNAMELEN). Should fail"""
+        testhost = HostTracker(name=u'testshost')
+        with raises_exact(errors.ValidationError()):
```

Comment:
If you are using raises_exact, you need to provide the exact exception, for instance:
errors.ValidationError(name='ipamaxhostnamelength', error=u'must be at least {}'.format(MAXHOSTNAMELEN))

Reply:
I couldn't find what exception is this going to raise as I suppose this is not hostname validation error, but rather other restraint. Thus I wanted  to let runner raise the exact one and edit this snippet accordingly.